aip/firebase/3260.md

```diff
@@ -25,22 +33,96 @@ based on common project selection, except when a command involves multiple
+provide a single-letter alias that does not conflict with [defined global
+flags][global-flags].
+
+#### Common Flags
+
+`--app <appId>` - used to refer to a specific app within a Firebase project.
+
+`--force` - used to indicate that a command should proceed without additional
+prompting regarding potentially destructive actions. For example, deploying
+Cloud Functions with `--force` will automatically delete functions, while
+without the user is prompted.
+
+`-o, --output <path>` - this flag **may** be used to allow convenient output of
+a serialized response to a local file. If the output is JSON, it **must**
+output equivalent JSON to `--json` without wrapped CLI metadata. If the command
+fetches a deployable resource, it **must** fetch it in a deploy-compatible
+format. The string path **may** be optional if a clear default path exists. If
+a file already exists at the path, a confirmation prompt **must** be displayed
+unless a `--force` flag is also provided.
+
+### Common Command Types
+
+#### List Commands
 
-### Naming Commands
+Commands that return a list of like resources **must** use a `:list` suffix
+(e.g. `projects:list`).
 
-- Commands **must** use parameters for arguments.
-- Commands **must** use a `:list` for listing multiple resources.
+List commands **must not** require manual pagination (e.g. a `--page-token`
+flag) and will usually recursively paginate all results by default. List
+commands **may** include a `--limit <int>` flag when a natural ordering exists,
+such as a reverse chronological list of releases. A `--limit 0` flag indicates
+that all results should be fetched and returned.
+
+The JSON output of a list command **must** be a concatenated array of all
+fetched resources in their original wire format.
+
+#### Clone Commands
+
+If a command's purpose is to copy information from one resource to another, it
+**must** be suffixed with `:clone` and **must** take a `from` and `to`
+argument:
+
+    firebase example:clone <fromProject> <toProject>
+
+To avoid confusion and accidental misuse, clone commands **must** explicitly
+require both arguments and **must not** rely on project selection to populate
+either.
 
 ### Deployment
 
 Behavior that takes local state (e.g. a rules file) and applies it to the live
 operation of a Firebase project **must** integrate with the `deploy` command.
 
-## Changelog
+#### Scoped Deployments
+
+If a service has multiple deployable resources, it **may** allow specifying
+which resources are deployed through colon namespacing, for example:
+
+    firebase deploy --only functions:func1,functions:func2,hosting:site3
+
+Colon namespacing **may** be used to identify one of many like resources (as
+with functions above) or to identify specific resource type (e.g.
+`firestore:indexes` vs. `firestore:rules`).
+
+If the service integrates with [deploy targets][deploy-targets], the targets
+**must** be addressable via colon namespacing.
+
+[global-flags]:
```

Comment:
I believe the Markdown parser is not accepting of this. You can get around it by adding `<!-- prettier-ignore-start -->` and `<!-- prettier-ignore-end -->` around it. (I usually just put the entirely of the link section in such a block whenever any one of them has this problem.)